cohorts/load.py

```diff
@@ -196,86 +410,206 @@ def _load_single_sample_variants(self, sample_idx, file_format_funcs, variant_ty
+        self.save_to_cache(merged_variants, self.cache_names["variant"], sample.id, cached_file_name)
 
         return merged_variants
 
     def load_effects(self, only_nonsynonymous=False, variant_type="snv", merge_type="union"):
         sample_effects = {}
-        for i, sample_id in enumerate(self.sample_ids):
+        for sample in self.iter_paired_samples():
             effects = self._load_single_sample_effects(
-                i, self.variant_type_to_format_funcs, only_nonsynonymous, variant_type, merge_type)
-            sample_effects[sample_id] = effects
+                sample, only_nonsynonymous, variant_type, merge_type)
+            sample_effects[sample.id] = effects
         return sample_effects
 
-    def _load_single_sample_effects(self, sample_idx, file_format_funcs,
-                                    only_nonsynonymous, variant_type, merge_type):
-        sample_id = self.sample_ids[sample_idx]
-
+    def _load_single_sample_effects(self, sample, only_nonsynonymous,
+                                    variant_type, merge_type):
         cached_file_name = "%s-%s-effects.pkl" % (variant_type, merge_type)
         if only_nonsynonymous:
-            cached = self.load_from_cache(self.cache_names["nonsynonymous_effect"], sample_id, cached_file_name)
+            cached = self.load_from_cache(self.cache_names["nonsynonymous_effect"], sample.id, cached_file_name)
         else:
-            cached = self.load_from_cache(self.cache_names["effect"], sample_id, cached_file_name)
+            cached = self.load_from_cache(self.cache_names["effect"], sample.id, cached_file_name)
         if cached is not None:
             return cached
 
         variants = self._load_single_sample_variants(
-            sample_idx, self.variant_type_to_format_funcs[variant_type], variant_type, merge_type)
+            sample, variant_type, merge_type)
         effects = variants.effects()
         nonsynonymous_effects = EffectCollection(
             effects.drop_silent_and_noncoding().top_priority_effect_per_variant().values())
 
-        self.save_to_cache(effects, self.cache_names["effect"], sample_id, cached_file_name)
-        self.save_to_cache(nonsynonymous_effects, self.cache_names["nonsynonymous_effect"], sample_id, cached_file_name)
+        self.save_to_cache(effects, self.cache_names["effect"], sample.id, cached_file_name)
+        self.save_to_cache(nonsynonymous_effects, self.cache_names["nonsynonymous_effect"], sample.id, cached_file_name)
 
         if only_nonsynonymous:
             return nonsynonymous_effects
         return effects
 
-    def load_neoantigens(self, variant_type="snv", merge_type="union",
-                         epitope_lengths=[8, 9, 10, 11], ic50_cutoff=500,
-                         process_limit=10, max_file_records=None):
-        assert self.hla_alleles is not None, "Cannot predict neoantigens without HLA alleles"
+    def load_cufflinks(self, filter_ok=True):
```

Comment:
Sure, I think `load.py` is ripe for splitting up, though I generally like to make file splits separate PRs for ease of code reviewing.